### ISSUE TITLE
2016 Talk 2: Jason

### ISSUE THREAD

**Issue by samth** [wonks/ICFP_rehearsal_feedback#14]

**Comment by rrnewton** [comment]
By "a lot more square brackets than we're used to", did you mean "curly braces"?  What was that alluding to?

**Comment by david-christiansen** [comment]
Why does `(member x d o)` not have an RHS in the `conde`? Or rather, why a different number of RHSes?

**Comment by david-christiansen** [comment]
What is the "diff2" style of constraint? Perhaps the audience will know.

**Comment by david-christiansen** [comment]
I don't recall you pointing out that `_.0` is the syntax for fresh names. That might be good, if you didn't.

**Comment by david-christiansen** [comment]
The "many languages" slide is very nice.

**Comment by ccshan** [comment]
Make the "Macros . Functions" slide(s) fill the screen, so that it's easier to see the punctuation change in the middle.

**Comment by david-christiansen** [comment]
What is "CL reducer/synthesizer"?

**Comment by david-christiansen** [comment]
The fact that member returns the whole list might be non-obvious to non-Lispers.

**Comment by david-christiansen** [comment]
the `==constraint` slide is nice, but you hopped through them faster than I could read the text and the code together.

**Comment by ccshan** [comment]
Love your antiunification animation, but can you make it zoom in at the same time so that the resulting 3-in-1 definition fills the screen?

**Comment by david-christiansen** [comment]
The `make-invalid?` implementation slide doesn't have text like the `==constraint` slide did. Consistency would be nice, I think that without text is better.

**Comment by ccshan** [comment]
The rule that your slides should contain at most 5 bullet points (total across all pages) is now active.

**Comment by david-christiansen** [comment]
Is `!-` the same as the turnstile symbol?

**Comment by samth** [comment]
- What is the point of this talk? Apparently from the paper it seems to be about adding CLP to microkanren.
- Are you really claiming that this is more efficient than Prolog?
- Don't say that this is different than traditional LP without being _very_ sure.
- How much do you know about what people are used to?
- Fewer transitions, both that remove text or just add it.
- So much code that it's hard to follow. What do you want the audience to take away?
- You don't have to show your system producing ugly output.

**Comment by vollmerm** [comment]
This was an entertaining talk, but I'm still not sure what your main contribution is. Maybe you could signal when you transition from the "background of minikanren" part of the talk to the "new stuff" part.

**Comment by jsiek** [comment]
Too much use of incremental builds in the slides, that is, text popping up and/or going away.

**Comment by sabry** [comment]
What you are really doing seems to be "playing with miniKanren" and your talk just seems to wander around showing interesting little things. But the talk really lacks a crisp motivation and contribution. In answers to questions and "off script" you said some interesting things about combining constraint systems over different domains etc and having a lightweight way of experiment with customized solvers that could make a crisp motivation and structure to the talk.
